### PR TITLE
feat(google_bigquery): load google big query from base64 string of cr…

### PR DESCRIPTION
**Pull request by ArslanSaleem**
…edentials

### CODE REVIEW

**Comment by gventuri** [comment]
@ArslanSaleem there's a linting error!

File: examples/from_googlebigquery.py [hunk 1]
```diff
@@ -1,25 +1,57 @@
...
+
+# Convert JSON data to a string
+json_string = json.dumps(json_data, indent=2)
+encoded_bytes = base64.b64encode(json_string.encode("utf-8"))
+
+
 bigquery_connectors = GoogleBigQueryConnector(
     config={
-        "credentials_path": "credentials.json",
+        "credentials_base64": encoded_bytes,
```
Comment (gventuri) [inline examples/from_googlebigquery.py:46]: @ArslanSaleem so we don't support the credentials_path anymore? Or we want to keep that as an option as well?

Also, is there any way that we can make it optional and automatically figure that out from the local environment if the user has been authenticated on the google cloud cli?

File: pandasai/ee/connectors/google_big_query.py [hunk 2]
```diff
@@ -56,11 +64,15 @@ def _init_connection(self, config: GoogleBigQueryConnectorConfig):
...
-        )
+        if config.credentials_path:
+            self._engine = create_engine(
+                f"{config.dialect}://{config.projectID}/{config.database}",
+                credentials_path=config.credentials_path,
+            )
+        else:
+            self._engine = create_engine(
+                f"{config.dialect}://{config.projectID}/{config.database}?credentials_base64={config.credentials_base64}"
+            )
```
Comment (gventuri) [inline pandasai/ee/connectors/google_big_query.py:75]: Can we write a test case for this?

File: examples/from_googlebigquery.py (hunk 1 above)
Comment (ArslanSaleem) [reply examples/from_googlebigquery.py:46]: @gventuri that i think is also possible but it need to use google.cloud lib should i do it?

File: pandasai/ee/connectors/google_big_query.py (hunk 2 above)
Comment (ArslanSaleem) [reply pandasai/ee/connectors/google_big_query.py:75]: Done

File: examples/from_googlebigquery.py (hunk 1 above)
Comment (gventuri) [reply examples/from_googlebigquery.py:46]: How involved would it be? Like you mean we need to create a custom version without sqlalchemy?

**Pull request merged**